### PR TITLE
Mention list of reserved and unreserved chars

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/uri.ex [hunk 1]
```diff
@@ -29,6 +29,15 @@ defmodule URI do
...
   import Bitwise
 
+  format_charlist = fn [first | rest] ->
+    Enum.reduce(rest, <<?`, first, ?`>>, fn char, acc ->
+      acc <> <<?,, ?\s, ?`, char, ?`>>
+    end)
+  end
+
+  @reserved_characters ':/?#[]@!$&\'()*+,;='
+  @formatted_reserved_characters format_charlist.(@reserved_characters)
```
Comment (josevalim) [inline lib/elixir/lib/uri.ex]:
```suggestion
  @formatted_reserved_characters Enum.map_join(@reserved_characters, ", ", &"`#{&1}`")
```

File: lib/elixir/lib/uri.ex [hunk 2]
```diff
@@ -218,14 +227,16 @@ defmodule URI do
...
   end
 
   @doc """
   Checks if `character` is an unreserved one in a URI.
 
-  Unreserved characters are specified in
-  [RFC 3986, section 2.3](https://tools.ietf.org/html/rfc3986#section-2.3).
+  As specified in [RFC 3986, section 2.3](https://tools.ietf.org/html/rfc3986#section-2.3)
+  the following characters are unreserved:
+    - Alphabumeric characters: `A-Z`, `a-z`, `0-9`
```
Comment (josevalim) [inline lib/elixir/lib/uri.ex]:
```suggestion

    * Alphanumeric characters: `A-Z`, `a-z`, `0-9`
```

File: lib/elixir/lib/uri.ex [hunk 3]
```diff
@@ -218,14 +227,16 @@ defmodule URI do
...
+    - `~`, `_`, `-`
```
Comment (josevalim) [inline lib/elixir/lib/uri.ex]:
```suggestion
    * `~`, `_`, `-`
```

File: lib/elixir/lib/uri.ex (hunk 1 above)
Comment (eksperimental) [reply lib/elixir/lib/uri.ex]: this is printing the codepoint
> "`58`, `47`, `63`, `35`, `91`, `93`, `64`, `33`, `36`, `38`, `39`, `40`, `41`, `42`, `43`, `44`, `59`, `61`"

Comment (eksperimental) [reply lib/elixir/lib/uri.ex]: it should be
``` 
Enum.map_join(reserved_characters, ", ", &"`#{<<&1>>}`")
```

Comment (eksperimental) [reply lib/elixir/lib/uri.ex]: Just out of curiosity, is there another way to get the string representation of a codepoint?

Comment (eksperimental) [reply lib/elixir/lib/uri.ex]: updated

Comment (josevalim) [reply lib/elixir/lib/uri.ex]: Or: 

    Enum.map_join(reserved_characters, ", ", &<<?`, &1, ?`>>)

File: lib/elixir/lib/uri.ex [hunk 4]
```diff
@@ -218,14 +221,17 @@ defmodule URI do
...
-    character in ':/?#[]@!$&\'()*+,;='
+    character in @reserved_characters
   end
 
   @doc """
   Checks if `character` is an unreserved one in a URI.
 
-  Unreserved characters are specified in
-  [RFC 3986, section 2.3](https://tools.ietf.org/html/rfc3986#section-2.3).
+  As specified in [RFC 3986, section 2.3](https://tools.ietf.org/html/rfc3986#section-2.3)
```
Comment (fertapric) [inline lib/elixir/lib/uri.ex]:
```suggestion
  As specified in [RFC 3986, section 2.3](https://tools.ietf.org/html/rfc3986#section-2.3),
```

File: lib/elixir/lib/uri.ex [hunk 5]
```diff
@@ -29,6 +29,9 @@ defmodule URI do
 
   import Bitwise
 
+  @reserved_characters ':/?#[]@!$&\'()*+,;='
+  @formatted_reserved_characters Enum.map_join(@reserved_characters, ", ", &"`#{<<&1>>}`")
```
Comment (fertapric) [inline lib/elixir/lib/uri.ex]: Minor nitpick:

```suggestion
  @formatted_reserved_characters Enum.map_join(@reserved_characters, ", ", &<<?`, &1, ?`>>)
```

**Comment by fertapric** [comment]
Thanks @eksperimental!

**Pull request merged**